Build Proxy Server Capabilities

evan-m-jackson opened the pull request:
- Add 'GetCode' method to 'Header' class to return status code
- Add if/else statement in Program class to allow for proxy server capabilities
- Add 'ProxyResponse' class and unit tests

evan-m-jackson committed:
Build Proxy Server Capabilities

- Add 'GetCode' method to 'Header' class to return status code
- Add if/else statement in Program class to allow for proxy server capabilities
- Add 'ProxyResponse' class and unit tests

evan-m-jackson committed:
Add PUT and DELETE capabilities to 'ProxyResponse' class

evan-m-jackson committed:
Refactor ProxyResponse class and tests

- Refactor GetBody method
- Remove unnecessary test

evan-m-jackson committed:
Rename namespaces to match file paths

jchung722 commented on HTTPServerProject/HTTPServerProxy/Response.cs:
```diff
@@ -0,0 +1,124 @@
+                GetFullJSONContentType(headers);
+                if (body.StartsWith('{') && body.EndsWith('}') && body.Length > 2)
+                {
+                    Console.WriteLine(body);
```
Were these `Console.WriteLine`s just used for debugging? It's probably good to remove these if so

evan-m-jackson replied:
Agreed.

evan-m-jackson committed:
Add ToDo First ID capabilities and format

- Refactor initialized variables to begin with '_'
- Remove unneccessary usings in each class
- Rename Proxy test files
- Add First ID class to return first todo item's primary id
- Remove empty whitespaces

evan-m-jackson committed:
Remove dictionay paramter default in ResponsePath class

evan-m-jackson committed:
Refactor FirstID class

- Create RunToListRequest method that runs the WriteToDoListRequest method until the body string reads the correct string, which is the JSON object with the to-do ids

evan-m-jackson committed:
Refactor Proxy Response Class

- Create 'GetContentLength' method that returns the length of the response body from the Proxy Server request and implemented this method into the 'GetResponse' method.
- Create 'StatusWrapper' method that returns the status code based on the HTTP method. This allowed for the 'GetResponse' method's design to be cleaned up.

evan-m-jackson committed:
Remove unnecessary print statements
